src/Particle/Par_MapMesh2Particles.cpp

```diff
@@ -0,0 +1,261 @@
+#include "GAMER.h"
+
+#ifdef PARTICLE
+
+void Par_MapMesh2Particles ( const int lv, const int P, const double EdgeL[3],
+                             const double EdgeR[3], const int AttrSize3D, const real *Attr,
```

Comment:
@hyschive so I actually tried inputting `Attr3D` at first (because it did make more sense) but I kept getting compilation errors (which I don't remember what they were anymore, something to do with pointers).

Reply:
@jzuhone I have just tested the following changes and it seems to work fine.
- `Par_UpdateTracerParticle.cpp`: when calling `Par_MapMesh2Particles()`, replace `VelX` with `VelX+P*CUBE(VelSize)` (same for `VelY` and `VelZ`)
- `Par_MapMesh2Particles.cpp`:
  - Declare `Attr3D` by
  ```
  typedef real (*vla)[AttrSize3D][AttrSize3D];
  vla Attr3D = ( vla )Attr;  
  ```
  - Replace `Attr3D[ P ]` with `Attr3D` when accessing its values (e.g., `ParAttr[p] = Attr3D[ idx[2] ][ idx[1] ][ idx[0] ];`)

Could you help test it too? Thanks.

Reply:
This works now.